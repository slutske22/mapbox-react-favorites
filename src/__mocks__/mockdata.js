export const results = [
	{
		feature: {
			type: 'Feature',
			id: 2605201641,
			geometry: {
				type: 'Point',
				coordinates: [-122.40506798028946, 37.79323314223906],
			},
			properties: {
				type: 'Yes',
				name_script: 'Latin',
				sizerank: 15,
				iso_3166_1: 'US',
				maki: 'marker',
				class: 'building',
				name: 'Provident Credit Union',
				filterrank: 5,
				iso_3166_2: 'US-CA',
				tilequery: {
					distance: 35.730233343837874,
					geometry: 'point',
					layer: 'poi_label',
				},
			},
		},
		marker: {
			_anchor: 'center',
			_color: '#C0C0C0',
			_scale: 1,
			_draggable: false,
			_clickTolerance: 0,
			_isDragging: false,
			_state: 'inactive',
			_rotation: 0,
			_rotationAlignment: 'auto',
			_pitchAlignment: 'auto',
			_defaultMarker: true,
			_element: {},
			_offset: {
				x: 0,
				y: -14,
			},
			_popup: {
				options: {
					closeButton: false,
					maxWidth: '500px',
					offset: {
						top: [0, 0],
						'top-left': [0, 0],
						'top-right': [0, 0],
						bottom: [0, -38.1],
						'bottom-left': [9.545941546018392, -34.14594154601839],
						'bottom-right': [-9.545941546018392, -34.14594154601839],
						left: [13.5, -24.6],
						right: [-13.5, -24.6],
					},
				},
				_content: {},
				_lngLat: {
					lng: -122.40506798028946,
					lat: 37.79323314223906,
				},
				_pos: {
					x: 368.34141402403884,
					y: 416.8159472192084,
				},
				_trackPointer: false,
				_tip: {},
			},
			_lngLat: {
				lng: -122.40506798028946,
				lat: 37.79323314223906,
			},
			_pos: {
				x: 348,
				y: 420,
			},
			_originalTabIndex: null,
		},
	},
	{
		feature: {
			type: 'Feature',
			id: 38919209430,
			geometry: {
				type: 'Point',
				coordinates: [-122.40443363785744, 37.793506564333185],
			},
			properties: {
				type: 'Ngo',
				name_script: 'Latin',
				sizerank: 16,
				iso_3166_1: 'US',
				maki: 'marker',
				class: 'general',
				name: 'United Way Bay Area',
				filterrank: 5,
				iso_3166_2: 'US-CA',
				tilequery: {
					distance: 43.44949856257183,
					geometry: 'point',
					layer: 'poi_label',
				},
			},
		},
		marker: {
			_anchor: 'center',
			_color: '#C0C0C0',
			_scale: 1,
			_draggable: false,
			_clickTolerance: 0,
			_isDragging: false,
			_state: 'inactive',
			_rotation: 0,
			_rotationAlignment: 'auto',
			_pitchAlignment: 'auto',
			_defaultMarker: true,
			_element: {},
			_offset: {
				x: 0,
				y: -14,
			},
			_popup: {
				options: {
					closeButton: false,
					maxWidth: '500px',
					offset: {
						top: [0, 0],
						'top-left': [0, 0],
						'top-right': [0, 0],
						bottom: [0, -38.1],
						'bottom-left': [9.545941546018392, -34.14594154601839],
						'bottom-right': [-9.545941546018392, -34.14594154601839],
						left: [13.5, -24.6],
						right: [-13.5, -24.6],
					},
				},
				_content: {},
				_lngLat: {
					lng: -122.40443363785744,
					lat: 37.793506564333185,
				},
				_pos: null,
				_trackPointer: false,
			},
			_lngLat: {
				lng: -122.40443363785744,
				lat: 37.793506564333185,
			},
			_pos: {
				x: 512,
				y: 331,
			},
			_originalTabIndex: null,
		},
	},
	{
		feature: {
			type: 'Feature',
			id: 51045550220,
			geometry: {
				type: 'Point',
				coordinates: [-122.40427538752556, 37.79285268113],
			},
			properties: {
				type: 'Cafe',
				name_script: 'Latin',
				sizerank: 16,
				iso_3166_1: 'US',
				maki: 'cafe',
				class: 'food_and_drink',
				name: 'Starbucks',
				'category_zh-Hans': '咖啡馆',
				filterrank: 5,
				category_en: 'Cafe',
				iso_3166_2: 'US-CA',
				tilequery: {
					distance: 49.108604123437786,
					geometry: 'point',
					layer: 'poi_label',
				},
			},
		},
		marker: {
			_anchor: 'center',
			_color: '#C0C0C0',
			_scale: 1,
			_draggable: false,
			_clickTolerance: 0,
			_isDragging: false,
			_state: 'inactive',
			_rotation: 0,
			_rotationAlignment: 'auto',
			_pitchAlignment: 'auto',
			_defaultMarker: true,
			_element: {},
			_offset: {
				x: 0,
				y: -14,
			},
			_popup: {
				options: {
					closeButton: false,
					maxWidth: '500px',
					offset: {
						top: [0, 0],
						'top-left': [0, 0],
						'top-right': [0, 0],
						bottom: [0, -38.1],
						'bottom-left': [9.545941546018392, -34.14594154601839],
						'bottom-right': [-9.545941546018392, -34.14594154601839],
						left: [13.5, -24.6],
						right: [-13.5, -24.6],
					},
				},
				_content: {},
				_lngLat: {
					lng: -122.40427538752556,
					lat: 37.79285268113,
				},
				_pos: {
					x: 552.5742971898977,
					y: 558,
				},
				_trackPointer: false,
				_tip: {},
			},
			_lngLat: {
				lng: -122.40427538752556,
				lat: 37.79285268113,
			},
			_pos: {
				x: 553,
				y: 544,
			},
			_originalTabIndex: null,
		},
	},
	{
		feature: {
			type: 'Feature',
			id: 46161412540,
			geometry: {
				type: 'Point',
				coordinates: [-122.40477964282036, 37.79364433477238],
			},
			properties: {
				type: 'Cafe',
				name_script: 'Latin',
				sizerank: 16,
				iso_3166_1: 'US',
				maki: 'cafe',
				class: 'food_and_drink',
				name: 'Young’s Cafe',
				'category_zh-Hans': '咖啡馆',
				filterrank: 5,
				category_en: 'Cafe',
				iso_3166_2: 'US-CA',
				tilequery: {
					distance: 54.17661511339451,
					geometry: 'point',
					layer: 'poi_label',
				},
			},
		},
		marker: {
			_anchor: 'center',
			_color: '#C0C0C0',
			_scale: 1,
			_draggable: false,
			_clickTolerance: 0,
			_isDragging: false,
			_state: 'inactive',
			_rotation: 0,
			_rotationAlignment: 'auto',
			_pitchAlignment: 'auto',
			_defaultMarker: true,
			_element: {},
			_offset: {
				x: 0,
				y: -14,
			},
			_popup: {
				options: {
					closeButton: false,
					maxWidth: '500px',
					offset: {
						top: [0, 0],
						'top-left': [0, 0],
						'top-right': [0, 0],
						bottom: [0, -38.1],
						'bottom-left': [9.545941546018392, -34.14594154601839],
						'bottom-right': [-9.545941546018392, -34.14594154601839],
						left: [13.5, -24.6],
						right: [-13.5, -24.6],
					},
				},
				_content: {},
				_lngLat: {
					lng: -122.40477964282036,
					lat: 37.79364433477238,
				},
				_pos: null,
				_trackPointer: false,
			},
			_lngLat: {
				lng: -122.40477964282036,
				lat: 37.79364433477238,
			},
			_pos: {
				x: 423,
				y: 286,
			},
			_originalTabIndex: null,
		},
	},
	{
		feature: {
			type: 'Feature',
			id: 2605201601,
			geometry: {
				type: 'Point',
				coordinates: [-122.4052195250988, 37.7928664583074],
			},
			properties: {
				type: 'Yes',
				name_script: 'Latin',
				sizerank: 15,
				iso_3166_1: 'US',
				maki: 'marker',
				class: 'building',
				name: '650 California Street',
				filterrank: 5,
				name_en: '650 California Street',
				iso_3166_2: 'US-CA',
				tilequery: {
					distance: 58.4454978763615,
					geometry: 'point',
					layer: 'poi_label',
				},
			},
		},
		marker: {
			_anchor: 'center',
			_color: '#C0C0C0',
			_scale: 1,
			_draggable: false,
			_clickTolerance: 0,
			_isDragging: false,
			_state: 'inactive',
			_rotation: 0,
			_rotationAlignment: 'auto',
			_pitchAlignment: 'auto',
			_defaultMarker: true,
			_element: {},
			_offset: {
				x: 0,
				y: -14,
			},
			_popup: {
				options: {
					closeButton: false,
					maxWidth: '500px',
					offset: {
						top: [0, 0],
						'top-left': [0, 0],
						'top-right': [0, 0],
						bottom: [0, -38.1],
						'bottom-left': [9.545941546018392, -34.14594154601839],
						'bottom-right': [-9.545941546018392, -34.14594154601839],
						left: [13.5, -24.6],
						right: [-13.5, -24.6],
					},
				},
				_content: {},
				_lngLat: {
					lng: -122.4052195250988,
					lat: 37.7928664583074,
				},
				_pos: {
					x: 325.5273637249197,
					y: 413.90078889628944,
				},
				_trackPointer: false,
				_tip: {},
			},
			_lngLat: {
				lng: -122.4052195250988,
				lat: 37.7928664583074,
			},
			_pos: {
				x: 309,
				y: 540,
			},
			_originalTabIndex: null,
		},
	},
];

export const favorites = results[0];
