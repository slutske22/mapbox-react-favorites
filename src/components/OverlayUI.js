import React from 'react';
import { GrFavorite } from 'react-icons/gr';
import { ImCross } from 'react-icons/im';
import './OverlayUI.scss';

const OverlayUI = ({ results, favorites, setFavorites }) => {
	return (
		<div className="overlay-ui-container">
			<div className="title-container card">
				<h3>
					<GrFavorite style={{ marginRight: '10px' }} />
					Favorite Keeper
				</h3>
				{results.length ? (
					<p>Click one of the result markers to add it to your favorites</p>
				) : (
					<p>Click on the map to see points of interest in the area.</p>
				)}
			</div>
			<div className="favorites-container card">
				<h3>Favorite Places</h3>
				{!favorites.length ? (
					<p>Choose some favorites!</p>
				) : (
					<ul>
						{favorites.map((favorite) => {
							const {
								id,
								properties: { name, category_en },
							} = favorite.feature;
							return (
								<li>
									<h5>{name}</h5>
									{category_en && <p>{category_en}</p>}
									<button>
										<ImCross size={18} />
									</button>
								</li>
							);
						})}
					</ul>
				)}
			</div>
		</div>
	);
};

export default OverlayUI;
