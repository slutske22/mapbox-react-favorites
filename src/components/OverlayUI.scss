.overlay-ui-container {
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	pointer-events: none;

	& > * {
		pointer-events: auto;
	}

	h3 {
		display: inline;
		margin: 0;
	}

	p {
		margin-bottom: 0;
	}

	.title-container {
		width: auto;
	}

	.favorites-container {
		position: absolute;
		top: 0px;
		right: 0px;
		width: 25%;
		min-width: 300px;
		max-width: 400px;
	}
}
