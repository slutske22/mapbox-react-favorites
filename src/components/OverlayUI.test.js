import React from 'react';
import OverlayUI from './OverlayUI';
import { mount } from 'enzyme';
import * as mockdata from '../__mocks__/mockdata';

describe('The OverlayUI component', () => {
	it('Renders consistently', () => {
		const wrapper = mount(
			<OverlayUI results={mockdata.results} favorites={mockdata.favorites} />
		).html();
		expect(wrapper).toMatchSnapshot();
	});

	it('Renders the "Click the map" message when there are no results present', () => {
		const wrapper = mount(
			<OverlayUI results={[]} favorites={mockdata.favorites} />
		);
		expect(wrapper.find('#click-the-map').exists()).toBeTruthy();
	});

	it('Renders the "Select a result" message when there are results present', () => {
		const wrapper = mount(
			<OverlayUI results={mockdata.results} favorites={mockdata.favorites} />
		);
		expect(wrapper.find('#select-a-result').exists()).toBeTruthy();
		expect(wrapper.find('#clear-results').exists()).toBeTruthy();
	});

	it('Calls the clear results method when the user clicks the "Clear results" button', () => {
		const setResults = jest.fn();
		const wrapper = mount(
			<OverlayUI
				results={mockdata.results}
				favorites={mockdata.favorites}
				setResults={setResults}
			/>
		);
		wrapper.find('#clear-results').simulate('click');
		expect(setResults).toHaveBeenCalledWith([]);
	});

	it('Displays a list item for every favorite', () => {
		const wrapper = mount(
			<OverlayUI results={[]} favorites={mockdata.results} />
		);

		mockdata.results.forEach((item) => {
			expect(
				wrapper.find(`#favorite-li-${item.feature.id}`).exists()
			).toBeTruthy();
		});
	});

	it('Calls the setFavorites function when the user clicks to remove a favorite', () => {
		const favorites = mockdata.results;
		const newFavorites = [...mockdata.results];
		newFavorites.shift();
		const setFavorites = jest.fn();

		const wrapper = mount(
			<OverlayUI
				results={[]}
				favorites={favorites}
				setFavorites={setFavorites}
			/>
		);

		wrapper
			.find(`#favorite-li-${mockdata.results[0].feature.id} button`)
			.simulate('click');

		expect(setFavorites).toHaveBeenCalledWith(newFavorites);
	});
});
