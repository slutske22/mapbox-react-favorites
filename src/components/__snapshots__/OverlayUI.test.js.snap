// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The OverlayUI component Renders consistently 1`] = `"<div class=\\"overlay-ui-container\\"><div class=\\"title-container card\\"><h3><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" style=\\"margin-right: 10px;\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M1,8.4 C1,4 4.5,3 6.5,3 C9,3 11,5 12,6.5 C13,5 15,3 17.5,3 C19.5,3 23,4 23,8.4 C23,15 12,21 12,21 C12,21 1,15 1,8.4 Z\\"></path></svg>Favorite Keeper</h3><p id=\\"select-a-result\\">Click one of the search result markers to add it to your favorites</p><button id=\\"clear-results\\">Clear Search Results</button></div><div class=\\"favorites-container card\\"><h3>Favorite Places</h3><p id=\\"click-the-map\\">Choose some favorites!</p></div></div>"`;
